### PR TITLE
Allow deletion of module from ModuleSelect

### PR DESCRIPTION
**Pull request by sciffany**
## Context
#2139

## Implementation
I changed `removeModule` in `TimetableContent.tsx` such that it accepts moduleCode type instead of ModuleWithColor.

With this, `removeModule` can be called with ease from both of its children, `ModulesSelect` and `TimetableModulesTable`

In ModulesSelect, when a module has been added, a Trash icon is added beside the `Added` badge. Clicking this calls `removeModule` and removes the module from the semester.

image

image

Edited interface
image

## Other Information

### CODE REVIEW

**Comment by sciffany** [comment]
@taneliang , I've done some changes. Some of the tests don't pass. 
image

Are they my fault?

**Comment by ZhangYiJiang** [comment]
Yeah, that design needs work. 

The ones you screenshotted are just warnings. Open the show more button and look for the actual errors, or just run `yarn lint:code` locally to see the errors

File: website/src/views/timetable/TimetableContent.tsx [hunk 1]
```diff
@@ -182,16 +182,15 @@ class TimetableContent extends React.Component<Props, State> {
...
 
-  removeModule = (module: ModuleWithColor) => {
+  removeModule = (moduleCode: ModuleCode) => {
     // Save the index of the module before removal so the tombstone can be inserted into
     // the correct position
-    const index = this.addedModules().findIndex(
-      ({ moduleCode }) => moduleCode === module.moduleCode,
-    );
-    this.props.removeModule(this.props.semester, module.moduleCode);
+    const index = this.addedModules().findIndex(({ moduleCode }) => moduleCode === moduleCode);
```
Comment (taneliang) [inline website/src/views/timetable/TimetableContent.tsx]: This line is causing lint to fail 
image

Comment (ZhangYiJiang) [reply website/src/views/timetable/TimetableContent.tsx]: By the way @sciffany you should try to enable ESLint integration or checks when developing, it'll flag these errors for you as you code. Feel free to talk to us on Telegram if you have problem setting them up

**Comment by sciffany** [comment]
image

**Comment by ZhangYiJiang** [comment]
Moved the negative margins to the container of the button instead of on the button itself, which seems to work better. Also it's generally a bad idea to not use units. The lint rule you're triggering is about not putting units on `0` specifically, not about removing units everywhere. 

Also while tabbing technically works, the tab order is not very intuitive (the first thing you select when you tab is the dropdown menu itself, and it doesn't matter which item you've selected), but fixing that is probably not worth the effort

File: website/src/views/timetable/ModulesSelect.tsx [hunk 2]
```diff
@@ -155,22 +155,25 @@ export class ModulesSelectComponent extends React.Component<Props, State> {
...
-                        onClick={() => {
-                          this.props.onRemoveModule(module.moduleCode);
-                        }}
-                      >
-                        <Trash className={styles.actionIcon} />{' '}
-                      </button>
-                    </Tooltip>
-                    <span className="badge badge-info">Added</span>
-                  </div>
+                  <>
```
Comment (taneliang) [inline website/src/views/timetable/ModulesSelect.tsx]: Is this fragment necessary? It only contains one div

**Pull request merged**